test/integration/targets/want_json_modules_posix/library/helloworld.py

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/python
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+# WANT_JSON
+
+import json
+import sys
+
+try:
+    with open(sys.argv[1], 'r') as f:
+        data = json.load(f)
+except (IOError, OSError, IndexError):
+    print(json.dumps(dict(msg="No argument file provided", failed=True)))
+    sys.exit(1)
```

Comment:
Since this module includes an error check for no argument file, I'd include testing that in the integration test. Just execute it as a script instead of a module to verify it fails as expected.

Reply:
Addressed in the latest commit